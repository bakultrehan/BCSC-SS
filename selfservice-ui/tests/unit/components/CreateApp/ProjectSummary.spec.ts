import { shallowMount, mount } from '@vue/test-utils';
import ProjectSummary from '@/components/CreateApp/ProjectSummary.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('ProjectSummary.vue', () => {
  const store = new Vuex.Store({
    modules: {
      ProjectInfoModule: {
        namespaced: true,
        state: {},
        getters: {
          getSingleProjectInfo: jest.fn(() => {
            return {
              description: 'test discription',
              id: 12,
              organizationName: 'new org',
              projectName: 'new project',
              users: [
                {
                  email: 'e@e.com',
                  firstName: 'fname',
                  id: 1,
                  lastName: 'lname',
                  phone: '09870987654321',
                  role: 2
                },
                {
                  email: 'bb@test.com',
                  firstName: 'john',
                  id: 17,
                  lastName: 'bb',
                  phone: '987654321',
                  role: 3
                },
                {
                  email: 'user@yopmail.com',
                  firstName: 'John',
                  id: 1,
                  lastName: 'Doe',
                  phone: '',
                  role: 1
                }
              ]
            };
          }),
          getFinalProjectSubmissionStatus: jest.fn(),
          errorStatus: jest.fn()
        },
        actions: {
          loadSingleProjectInfo: jest.fn()
        }
      },
      TechnicalReqModule: {
        namespaced: true,
        state: {},
        getters: {
          getTechnicalReq: jest.fn(() => {
            return {
              clientUri: 'app.com',
              id: 15,
              encryptedResponseAlg: 'RS256',
              jwksUri: 'jwks2.com',
              noOfTestAccount: 3,
              noteTestAccount: '',
              projectId: 12,
              redirectUris: ['test.com'],
              scopePackageId: 3,
              signedResponseAlg: 'RS256'
            };
          }),
          errorStatus: jest.fn()
        },
        actions: {
          loadTechnicalReqDetails: jest.fn()
        }
      },
      PackageAndTestModule: {
        namespaced: true,
        state: {},
        getters: {
          successStatus: jest.fn(),
          getPackageList: jest.fn(() => {
            return [
              {
                claimNames: ['Given name', 'Surname'],
                description: 'This package contains the following data:',
                id: 1,
                packageName: 'Package 1'
              }
            ];
          })
        },
        actions: {
          loadPackage: jest.fn()
        }
      },
      SharedModule: {
        namespaced: true,
        state: {},
        getters: {
          isRedirectFromSummaryPage: jest.fn()
        },
        actions: {
          redirectFromSummaryPage: jest.fn()
        }
      },
      ClientIdModule: {
        namespaced: true,
        state: {},
        getters: {
          isLoading: jest.fn(),
          getApiData: jest.fn(() => {
            return {
              oidcConfig: {
                clientId: 'abcd',
                clientSecret: 'test'
              },
              testUserAccounts: [
                {
                  userName: 'test user',
                  idKey: 'test id key'
                },
                {
                  userName: 'test user',
                  idKey: 'test id key'
                }
              ]
            };
          }),
          errorStatus: jest.fn()
        },
        actions: {
          getClientIdDetails: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return mount(ProjectSummary, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const projectSummary = mountFunction({});
    expect(projectSummary.element).toMatchSnapshot();
  });

  it('show modal on click', () => {
    const projectSummary = mountFunction({});

    projectSummary.setData({ isLoading: false });
    const button = projectSummary.find('.submit-package');
    projectSummary.vm.$on('action-btn:clicked', jest.fn());
    button.trigger('click');
    expect(projectSummary.element).toMatchSnapshot();
  });
});

describe('ProjectSummary.vue', () => {
  it('renders props when passed', () => {
    const store = new Vuex.Store({
      modules: {
        ProjectInfoModule: {
          namespaced: true,
          state: {},
          getters: {
            getSingleProjectInfo: jest.fn(),
            errorStatus: jest.fn()
          },
          actions: {
            loadSingleProjectInfo: jest.fn()
          }
        },
        TechnicalReqModule: {
          namespaced: true,
          state: {},
          getters: {
            getTechnicalReq: jest.fn(),
            errorStatus: jest.fn()
          },
          actions: {
            loadTechnicalReqDetails: jest.fn()
          }
        },
        PackageAndTestModule: {
          namespaced: true,
          state: {},
          getters: {
            successStatus: jest.fn(),
            getPackageList: jest.fn()
          },
          actions: {
            loadPackage: jest.fn()
          }
        },
        SharedModule: {
          namespaced: true,
          state: {},
          getters: {
            isRedirectFromSummaryPage: jest.fn()
          },
          actions: {
            redirectFromSummaryPage: jest.fn()
          }
        }
      }
    });
    const projectSummary = shallowMount(ProjectSummary, {
      vuetify,
      store,
      mocks: { $t: jest.fn(() => {}) } // tslint:disable-line
    });
    expect(projectSummary.element).toMatchSnapshot();
  });
});
