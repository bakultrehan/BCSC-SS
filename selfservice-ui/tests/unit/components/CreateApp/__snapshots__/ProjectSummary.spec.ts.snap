// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSummary.vue renders props when passed 1`] = `
<div>
  <loading-stub />
   
  <div
    class="text-center"
  >
    <v-dialog-stub
      class="text-left"
      closedelay="0"
      contentclass=""
      maxwidth="none"
      opendelay="0"
      origin="center center"
      persistent="true"
      retainfocus="true"
      transition="dialog-transition"
      width="70%"
    >
      <v-card-stub
        loaderheight="4"
        tag="div"
      >
        <v-card-title-stub
          class="bc-subtitle padding-0"
          primary-title=""
          primarytitle=""
        >
          
          
        
        </v-card-title-stub>
         
        <v-card-text-stub
          class="text-left"
        >
          <div />
        </v-card-text-stub>
         
        <v-divider-stub />
         
        <v-card-actions-stub>
          <v-spacer-stub />
           
          <button-stub
            aria-label="Back Button"
            color=""
            data-test-id="btn-cancel-project-summary"
            name="button"
            secondary=""
            type="button"
          >
            
          </button-stub>
           
          <button-stub
            class="white--text submit-package ml-6"
            color=""
            data-test-id="btn-submit-project-summary"
            depressed=""
            name="button"
            type="button"
          >
            
          </button-stub>
        </v-card-actions-stub>
      </v-card-stub>
    </v-dialog-stub>
  </div>
</div>
`;

exports[`ProjectSummary.vue renders props when passed with gettors 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="layout d-flex align-center justify-center row wrap"
    >
      <div
        class="mt-10"
        col="4"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-linear loading v-progress-linear--rounded theme--light"
          role="progressbar"
          style="height: 6px;"
        >
          <div
            class="v-progress-linear__background primary"
            style="opacity: 0.3; left: 0%; width: 100%;"
          />
          <div
            class="v-progress-linear__buffer"
          />
          <div
            class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active"
          >
            <div
              class="v-progress-linear__indeterminate long primary"
            />
            <div
              class="v-progress-linear__indeterminate short primary"
            />
          </div>
        </div>
         
        <span />
      </div>
    </div>
  </div>
   
  <div
    class="text-center"
  >
    <div
      class="v-dialog__container text-left"
      role="dialog"
    >
      <div
        class="v-dialog__content"
        role="document"
        style="z-index: 0;"
      >
        <div
          class="v-dialog v-dialog--persistent"
          style="width: 70%; display: none;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectSummary.vue show modal on click 1`] = `
<div>
  <div
    class="row"
  >
    <!---->
     
    <!---->
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="v-list-item__content text-left padding-0"
      />
    </div>
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="mx-auto mb-5 v-card v-sheet theme--light"
      >
        <!---->
         
        <!---->
         
        <header
          class="bc-subtitle-gold-1 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-card__title"
            >
              
            </div>
          </div>
        </header>
         
        <div
          class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
          role="list"
        >
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content pr-30"
            >
              
        
        
              <span
                class="small-hint"
              />
            </div>
             
            <div
              class="v-list-item__content align-end pointer client-id-copy"
            >
              <code
                class="code-snippet"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mr-1 mdi mdi-content-copy theme--light"
                  style="font-size: 16px;"
                />
                
          new client
        
              </code>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--light"
            role="separator"
          />
           
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content pr-30"
            >
              
        
        
              <span
                class="small-hint"
              />
            </div>
             
            <div
              class="v-list-item__content align-end pointer d-block"
            >
              <code
                class="code-snippet"
              >
                <div
                  class="float-left"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mr-2 mdi mdi-content-copy theme--light"
                    style="font-size: 16px;"
                  />
                </div>
                 
                <span>
                  new secret
                </span>
              </code>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--light"
            role="separator"
          />
           
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content align-self-start pr-30"
            >
              <div>
                
          
          
                <span
                  class="v-tooltip v-tooltip--bottom"
                >
                  <div
                    class="v-tooltip__content"
                    style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"
                  />
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-help-circle theme--light"
                  />
                </span>
              </div>
               
              <span
                class="small-hint"
              />
            </div>
             
            <div
              class="v-list-item__content align-self-start"
            >
              <div
                class="v-data-table v-data-table--dense theme--light"
              >
                <div
                  class="v-data-table__wrapper"
                >
                  <table>
                    <thead>
                      <tr>
                        <th
                          class="text-left"
                        >
                          
                  
                
                        </th>
                         
                        <th
                          class="text-left"
                        >
                          
                  
                
                        </th>
                         
                        <th
                          class="text-left"
                        >
                          
                  
                
                        </th>
                      </tr>
                    </thead>
                     
                    <tbody />
                  </table>
                </div>
              </div>
               
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="container"
        >
          <div
            class="layout d-flex align-center justify-center row wrap"
          >
            <div
              class="mt-10"
              col="4"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                class="v-progress-linear loading v-progress-linear--rounded theme--light"
                role="progressbar"
                style="height: 6px;"
              >
                <div
                  class="v-progress-linear__background primary"
                  style="opacity: 0.3; left: 0%; width: 100%;"
                />
                <div
                  class="v-progress-linear__buffer"
                />
                <div
                  class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active"
                >
                  <div
                    class="v-progress-linear__indeterminate long primary"
                  />
                  <div
                    class="v-progress-linear__indeterminate short primary"
                  />
                </div>
              </div>
               
              <span />
            </div>
          </div>
        </div>
         
        <!---->
      </div>
    </div>
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="mt-5 v-card v-sheet theme--light red-border"
      >
        <header
          class="bc-subtitle-2 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-card__title"
            >
              
      
      
              <div
                class="spacer"
              />
               
              <i
                class="v-icon notranslate ml-3 v-icon--link mdi mdi-pencil theme--dark"
                role="button"
                style="font-size: 16px;"
                tabindex="0"
              />
            </div>
          </div>
        </header>
         
        <div
          class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
          role="list"
        >
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content align-self-start pr-30"
            >
              
          
          
              <span
                class="small-hint pad-50"
              />
            </div>
             
            <div
              class="v-list-item__content align-end"
            >
              <div>
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mr-1 mdi mdi-account theme--light"
                  style="font-size: 16px;"
                />
                
            
            
          
              </div>
               
              <div
                class="ml-6"
              >
                
            
          
              </div>
               
              <div
                class="ml-6"
              >
                
            
          
              </div>
            </div>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="mt-5 v-card v-sheet theme--light red-border"
      >
        <header
          class="bc-subtitle-2 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-card__title"
            >
              
      
      
              <i
                class="v-icon notranslate ml-3 v-icon--link mdi mdi-pencil theme--dark"
                role="button"
                style="font-size: 16px;"
                tabindex="0"
              />
            </div>
          </div>
        </header>
         
        <div
          class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
          role="list"
        >
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content pr-30"
            >
              
        
        
              <span
                class="small-hint pad-50"
              />
            </div>
             
            <div
              class="v-list-item__content align-end"
            >
              <div>
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mr-1 mdi mdi-link theme--light"
                  style="font-size: 16px;"
                />
                
          app.com
        
              </div>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--light"
            role="separator"
          />
           
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content align-self-start pr-30"
            >
              
        
        
              <span
                class="small-hint pad-50"
              />
            </div>
             
            <div
              class="v-list-item__content align-end"
            >
              <div>
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mr-1 mdi mdi-link theme--light"
                  style="font-size: 16px;"
                />
                
          test.com
        
              </div>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--light"
            role="separator"
          />
           
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content pr-30"
            >
              
        
        
              <span
                class="small-hint pad-50"
              />
            </div>
             
            <div
              class="v-list-item__content align-end"
            >
              <!---->
            </div>
          </div>
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="mt-5 v-card v-sheet theme--light bc-disabled-section"
      >
        <header
          class="bc-subtitle-2 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-card__title"
            >
              
      
      
              <i
                class="v-icon notranslate ml-3 v-icon--link mdi mdi-pencil theme--dark"
                role="button"
                style="font-size: 16px;"
                tabindex="0"
              />
            </div>
          </div>
        </header>
         
        <div
          class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
          role="list"
        >
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content align-self-start pr-30"
            >
              
        
        
              <span
                class="small-hint pad-50"
              />
            </div>
             
            <div
              class="v-list-item__content align-end"
            >
              <!---->
               
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col col-12"
      flat=""
    />
     
    <div
      class="col col-12"
      flat=""
    >
      <div
        class="mx-auto mb-5 v-card v-sheet theme--light bc-disabled-section"
      >
        <header
          class="bc-subtitle-2 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-card__title"
            >
              
      
      
              <i
                class="v-icon notranslate ml-3 v-icon--link mdi mdi-pencil theme--dark"
                role="button"
                style="font-size: 16px;"
                tabindex="0"
              />
            </div>
          </div>
        </header>
         
        <div
          class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
          role="list"
        >
          <div
            class="v-list-item theme--light"
            role="listitem"
            tabindex="-1"
          >
            <div
              class="v-list-item__content pr-30"
            >
              
        
      
            </div>
             
            <div
              class="v-list-item__content align-end"
            >
              <div>
                <span
                  class="font-weight-bold"
                >
                  3
                </span>
                
          
        
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col col-12"
    >
      <div
        class="v-alert text-left text-left v-sheet theme--light v-alert--dense v-alert--outlined error--text"
        role="alert"
      >
        <div
          class="v-alert__wrapper"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate v-alert__icon mdi mdi-alert theme--light error--text"
          />
          <div
            class="v-alert__content"
          >
            
          </div>
        </div>
      </div>
       
      <!---->
    </div>
     
    <div
      class="col col-12"
    >
      <div
        class="mt-1 v-card v-card--flat v-sheet theme--light"
      >
        <hr
          aria-orientation="horizontal"
          class="v-divider theme--light"
          role="separator"
        />
         
        <div
          class="v-card__actions mt-2 py-2 px-0"
        >
          <div
            class="spacer"
          />
           
          <!---->
           
          <button
            class="btn white--text submit-package ml-6 v-btn v-btn--depressed theme--light v-size--default  primary  normal"
            data-test-id="btn-submit-summary"
            depressed=""
            name="button"
            type="button"
          >
            <span
              class="v-btn__content"
            >
              
            
          
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="text-center"
  >
    <div
      class="v-dialog__container text-left"
      role="dialog"
    >
      <div
        class="v-dialog__content"
        role="document"
        style="z-index: 0;"
      >
        <div
          class="v-dialog v-dialog--persistent"
          style="width: 70%; display: none;"
        />
      </div>
    </div>
  </div>
</div>
`;
